### PR TITLE
Test Julia release candidate

### PR DESCRIPTION
**Pull request by SylvainCorlay**
Reproducing the issue reported by @nbecker with julia 0.6.

cc @barche

### CODE REVIEW

File: deps/xtensor-julia/include/xtensor-julia/jlarray.hpp [hunk 1]
```diff
@@ -410,10 +410,16 @@ namespace cxx_wrap
...
+#if JULIA_VERSION_MAJOR == 0 && JULIA_VERSION_MINOR < 6
             return (jl_datatype_t*)apply_type(
                 jl_get_global(jl_current_module, jl_symbol("Array")),
                 jl_svec1(cxx_wrap::julia_type<T>())
             );
+#else
+            // TODO: workaround for Julia 0.6
+            std::string cmd = "Array{" + cxx_wrap::julia_type_name(cxx_wrap::julia_type<T>()) + "}";
+            return (jl_datatype_t*)jl_eval_string(cmd.c_str());
+#endif
```
Comment (SylvainCorlay) [inline deps/xtensor-julia/include/xtensor-julia/jlarray.hpp]: Attempt for a workaround for Julia 0.6.
This seems to confirm that the issue lies with what is returned.

**Comment by SylvainCorlay** [comment]
So the issue is that

In Julia < 0.6

```
> typeof(Array{Float64})
DataType
```

In Julia 0.6
```
> typeof(Array{Float64})
UnionAll
```

and
```
> isa(DataType, UnionAll)
false
```

**Comment by SylvainCorlay** [comment]
@nbecker we have a fix for this issue, but this will require the next CxxWrap release.

https://github.com/JuliaInterop/CxxWrap.jl/pull/59

**Comment by nbecker** [comment]
cool, thanks! Looking forward to trying it.

**Pull request merged**